edx_django_utils/hooks/triggers.py

```diff
@@ -35,10 +41,17 @@ def trigger_filter(trigger_name, *args, **kwargs):
     if not pipeline:
         return kwargs
 
+    result = kwargs
+
     if is_async:
-        result = run_pipeline(
-            pipeline, *args, raise_exception=True, **kwargs
-        )  # TODO: change to async call.
+        try:
+            result = run_pipeline.delay(pipeline, *args, raise_exception=True, **kwargs)
+        except (TypeError, EncodeError):
+            log.exception(
+                "An error ocurred in trigger_filter while executing `run pipeline` with arguments: %s, %s.",
+                str(args),
+                str(kwargs),
```

Comment:
maybe this won't be needed

Reply:
It makes no difference what the pipeline is at this point right?
I mean, we are sure the pipeline object is never the source of this error. It is always the serialization, right?

Reply:
Yes, it should always be serialization. I'll add it to the message!